client/laststand.lua

```diff
@@ -30,84 +30,90 @@ local function GetClosestPlayer()
-                    local weaponLabel = Lang:t('info.wep_unknown')
-                    local weaponName = Lang:t('info.wep_unknown')
-                    local weaponItem = QBCore.Shared.Weapons[killerWeapon]
-                    if weaponItem then
-                        weaponLabel = weaponItem.label
-                        weaponName = weaponItem.name
-                    end
-                    TriggerServerEvent("qb-log:server:CreateLog", "death", Lang:t('logs.death_log_title', { playername = GetPlayerName(-1), playerid = GetPlayerServerId(player) }), "red", Lang:t('logs.death_log_message', { killername = killerName, playername = GetPlayerName(player), weaponlabel = weaponLabel, weaponname = weaponName }))
-                    deathTime = 0
-                    OnDeath()
-                    DeathTimer()
-                end
-                Wait(1000)
-            end
-        end)
+local function playLastStandAnimation(ped)
+    if cache.vehicle then
+        lib.requestAnimDict("veh@low@front_ps@idle_duck")
+        TaskPlayAnim(ped, "veh@low@front_ps@idle_duck", "sit", 1.0, 8.0, -1, 1, -1, false, false, false)
     else
-        TaskPlayAnim(ped, lastStandDict, "exit", 1.0, 8.0, -1, 1, -1, false, false, false)
-        InLaststand = false
-        LaststandTime = 0
+        lib.requestAnimDict(lastStandDict)
+        TaskPlayAnim(ped, lastStandDict, lastStandAnim, 1.0, 8.0, -1, 1, -1, false, false, false)
     end
-    TriggerServerEvent("hospital:server:SetLaststandStatus", bool)
+end
+
+local function logPlayerKiller()
+    local ped = cache.ped
+    local player = cache.playerId
+    local killer_2, killerWeapon = NetworkGetEntityKillerOfPlayer(player)
+    local killer = GetPedSourceOfDeath(ped)
+    if killer_2 ~= 0 and killer_2 ~= -1 then killer = killer_2 end
+    local killerId = NetworkGetPlayerIndexFromPed(killer)
+    local killerName = killerId ~= -1 and GetPlayerName(killerId) .. " " .. "(" .. GetPlayerServerId(killerId) .. ")" or Lang:t('info.self_death')
+    local weaponLabel = Lang:t('info.wep_unknown')
+    local weaponName = Lang:t('info.wep_unknown')
+    local weaponItem = QBCore.Shared.Weapons[killerWeapon]
+    if weaponItem then
+        weaponLabel = weaponItem.label
+        weaponName = weaponItem.name
+    end
+    TriggerServerEvent("qb-log:server:CreateLog", "death", Lang:t('logs.death_log_title', { playername = GetPlayerName(cache.playerId), playerid = GetPlayerServerId(player) }), "red", Lang:t('logs.death_log_message', { killername = killerName, playername = GetPlayerName(player), weaponlabel = weaponLabel, weaponname = weaponName }))
+end
+
+local function countdownLastStand()
+    if LaststandTime - 1 <= 0 then
+        lib.notify({ description = Lang:t('error.bled_out'), type = 'error' })
+        endLastStand()
+        logPlayerKiller()
+        deathTime = 0
+        OnDeath()
+        DeathTimer()
+    else
+        LaststandTime -= 1
+        Config.DeathTime = LaststandTime
+    end
```

Comment:
I would consider swapping this around so that the condition appears simpler. It also has the benefit of maintaining the typical order of operation within the last stand.
```suggestion
    if LaststandTime - 1 > 0 then
        LaststandTime -= 1
        Config.DeathTime = LaststandTime
    else
        lib.notify({ description = Lang:t('error.bled_out'), type = 'error' })
        endLastStand()
        logPlayerKiller()
        deathTime = 0
        OnDeath()
        DeathTimer()
    end
```